### PR TITLE
Add set_learning_rate to OptimizationABC

### PR DESCRIPTION
**Pull request by mcgibbon**
## Summary
- Add `set_learning_rate(lr)` abstract method to `OptimizationABC` and implement on `Optimization` (updates all param groups) and `NullOptimization` (no-op)

## Test plan
- [x] New tests verify learning rate is updated and optimizer momentum state is preserved
- [x] Existing optimization tests pass

### CODE REVIEW

File: fme/core/test_optimization.py [hunk 1]
```diff
@@ -416,6 +416,150 @@ def test_sequential_scheduler_reload():
...
+    Optimization into another built with different parameter objects but
+    the same structure. This is the pattern used by LR tuning trials,
+    where we deepcopy a model and need the fork's optimizer to start
+    with the original's momentum.
+    """
+    torch.manual_seed(0)
+    model = nn.Linear(2, 2).to(fme.get_device())
+    x = torch.randn(10, 2).to(fme.get_device())
+
+    optimization = Optimization(
```
Comment (elynnwu) [inline fme/core/test_optimization.py]: [nit] add a helper function to construct Optimization to make code more compact
